numpydoc/tests/test_docscrape.py

```diff
@@ -796,11 +796,13 @@ class BadSection(object):
         pass
 
     with warnings.catch_warnings(record=True) as w:
+        warnings.filterwarnings('always', '', UserWarning)
```

Comment:
I am not sure how test runs avoid the need for a filter to work properly, this is more in line with what numpy does.